lib/usd/utils/util.cpp

```diff
@@ -147,43 +190,16 @@ updateInternalReferencesPath(const UsdPrim& oldPrim, const SdfPath& newPath)
     SdfChangeBlock changeBlock;
     for (const auto& p : oldPrim.GetStage()->Traverse()) 
     {
-        if(p.HasAuthoredReferences()) {
+        if(p.HasAuthoredReferences())
+        {
             auto primSpec = getPrimSpecAtEditTarget(p);
-            if (primSpec) {
-                for (const SdfReference &ref : primSpec->GetReferenceList().GetAddedOrExplicitItems()) {
-                    if (ref.IsInternal())
-                    {
-                        SdfPath finalPath;
-                        if(oldPrim.GetPath() == ref.GetPrimPath()) {
-                            finalPath = newPath;
-                        }
-                        else if(ref.GetPrimPath().HasPrefix(oldPrim.GetPath())) {
-                            finalPath = ref.GetPrimPath().ReplacePrefix(oldPrim.GetPath(), newPath);
-                        }
-
-                        if(finalPath.IsEmpty()){
-                            continue;
-                        }
-
-                        // clear the references first
-                        bool status = p.GetReferences().ClearReferences();
-                        if (!status) {
-                            return false;
-                        }
-
-                        // NOTE: currently there is no APIs either at 
-                        // Sdf or Usd level to help querying the "append" or 
-                        // "prepend" operations for the entries in the list. Therefore, UsdListPosition 
-                        // is always set by the default value UsdListPositionBackOfPrependList passed to
-                        // AddInternalReference
-
-                        // add the new internal reference
-                        status = p.GetReferences().AddInternalReference(finalPath);
-                        if (!status) {
-                            return false;
-                        }
-                    }
-                }
+            if (primSpec)
+            {
+                SdfReferencesProxy referencesList = primSpec->GetReferenceList();
+
+                // update append/prepend lists individually 
+                replaceReferenceItems(oldPrim, newPath, referencesList, SdfListOpTypeAppended);
+                replaceReferenceItems(oldPrim, newPath, referencesList, SdfListOpTypePrepended);
             }
```

Comment:
@ppt-adsk  We need to update each individual list separately.  For now, I am only concerned about append and prepend list Ops.   

As a side note, the word "list" could be a bit confusing. Each list edit operation ( e.g appen , prepend, etc.. ) are stored in std::vector like containers. 

> An SdfListProxy represents a single list of list editing operations, making it look like an STL vector (modeling a random access container and back  insertion sequence). 

https://github.com/PixarAnimationStudios/USD/blob/release/pxr/usd/sdf/listProxy.h